inst/include/mode/mode.hpp

```diff
@@ -13,6 +13,11 @@ namespace mode {
 template <typename T>
 typename mode::pars_type<T> mode_pars(cpp11::list pars);
 
+template <typename T>
+cpp11::sexp mode_info(const mode::pars_type<T>& pars) {
+  return R_NilValue;
```

Comment:
The version without a function body is a declaration (it allows template specialisations to be defined later). However, no default is provided and it will fail to compile (or really to link as there will be an unresolved symbol). We can't really have a case where a model does not provide any parameter handling so there is no default.

For the `mode_info` one, this is optional, and if no more specific specialisation is provided we'll use this version that returns an R `NULL`